src/rdkafka_admin.c

```diff
@@ -480,6 +571,15 @@ static void rd_kafka_admin_common_worker_destroy (rd_kafka_t *rk,
                 rd_kafka_enq_once_destroy(rko->rko_u.admin_request.eonce);
                 rko->rko_u.admin_request.eonce = NULL;
         }
+
+        if (rko->rko_u.admin_request.admin_fanout_parent) {
```

Comment:
Have you run the Admin API tests with helgrind or tsan enabled? Just to make sure there is no concurrency issues introduced.

Reply:
Yep, I've been running with tsan on my mac and it seems happy.